Bug: Ladybird fails to build on MacOS 14 using python script

### Note
I know the issue guidelines say "No build issues" but this seems like it might be an actual bug and not a build issue from my end as I am able to build with my suggested fix.

## General bug info
I was trying to build Ladybird commit `9ee9be720c40a052db202a0069f70174ffb7ddb8` (`master` at time of writting) on MacOS 14 Sonoma.
The build started but failed at step `[3401/3403] Building CXX object UI/AppKit/CMakeFiles/ladybird_impl.dir/Interface/TabController.mm.o` with the error:
```
FAILED: [code=1] UI/AppKit/CMakeFiles/ladybird_impl.dir/Interface/TabController.mm.o
/opt/homebrew/opt/llvm@20/bin/clang++  -I/Users/seppedelanghe/Documents/side/ladybird -I/Users/seppedelanghe/Documents/side/ladybird/Services -I/Users/seppedelanghe/Documents/side/ladybird/Libraries -I/Users/seppedelanghe/Documents/side/ladybird/Build/release/Lagom -I/Users/seppedelanghe/Documents/side/ladybird/Build/release/Lagom/Services -I/Users/seppedelanghe/Documents/side/ladybird/Build/release/Lagom/Libraries -I/Users/seppedelanghe/Documents/side/ladybird/UI/AppKit -I/Users/seppedelanghe/Documents/side/ladybird/Build/release/UI -isystem /Users/seppedelanghe/Documents/side/ladybird/Build/release/vcpkg_installed/arm64-osx-dynamic/include -O2 -g -DNDEBUG -std=c++23 -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -mmacosx-version-min=14.0 -fPIC -fcolor-diagnostics -march=native -Wall -Wextra -fno-exceptions -ffp-contract=off -Wcast-qual -Wformat=2 -Wimplicit-fallthrough -Wlogical-op -Wmissing-declarations -Wmissing-field-initializers -Wsuggest-override -Wno-invalid-offsetof -Wno-unknown-warning-option -Wno-unused-command-line-argument -Wpadded-bitfield -Werror -fconstexpr-steps=16777216 -Wmissing-prototypes -Wno-implicit-const-int-float-conversion -Wno-user-defined-literals -Wno-unqualified-std-cast-call -fstack-protector-strong -fstrict-flex-arrays=2 -Wno-maybe-uninitialized -Wno-shorten-64-to-32 -fsigned-char -ggnu-pubnames -O2 -g1 -Wno-expansion-to-defined -fobjc-arc -Wno-deprecated-anon-enum-enum-conversion -MD -MT UI/AppKit/CMakeFiles/ladybird_impl.dir/Interface/TabController.mm.o -MF UI/AppKit/CMakeFiles/ladybird_impl.dir/Interface/TabController.mm.o.d -o UI/AppKit/CMakeFiles/ladybird_impl.dir/Interface/TabController.mm.o -c /Users/seppedelanghe/Documents/side/ladybird/UI/AppKit/Interface/TabController.mm
/Users/seppedelanghe/Documents/side/ladybird/UI/AppKit/Interface/TabController.mm:97:27: error: no visible @interface for 'NSToolbar' declares the selector 'setAllowsDisplayModeCustomization:'
   97 |             [self.toolbar setAllowsDisplayModeCustomization:NO];
      |              ~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
ninja: build stopped: subcommand failed.
```

## Device & build info
#### Device:
- Macbook Pro M1 Pro
- MacOS 14.7.8

#### Build environment:
LLVM / clang 20.1.8 (installed with homebrew)
```
Homebrew clang version 20.1.8
Target: arm64-apple-darwin23.6.0
Thread model: posix
InstalledDir: /opt/homebrew/Cellar/llvm@20/20.1.8/bin
Configuration file: /opt/homebrew/etc/clang/arm64-apple-darwin23.cfg
System configuration file directory: /opt/homebrew/etc/clang
User configuration file directory: /Users/seppedelanghe/.config/clang
```

## Possible fix
I modified the file `UI/AppKit/Interface/TabController.mm` like this:

```diff
diff --git a/UI/AppKit/Interface/TabController.mm b/UI/AppKit/Interface/TabController.mm
index e9aeb93f4e..ec26647f5e 100644
--- a/UI/AppKit/Interface/TabController.mm
+++ b/UI/AppKit/Interface/TabController.mm
@@ -93,8 +93,8 @@ - (instancetype)init
         self.toolbar = [[NSToolbar alloc] initWithIdentifier:TOOLBAR_IDENTIFIER];
         [self.toolbar setDelegate:self];
         [self.toolbar setDisplayMode:NSToolbarDisplayModeIconOnly];
-        if (@available(macOS 15, *)) {
-            [self.toolbar setAllowsDisplayModeCustomization:NO];
+        if ([self.toolbar respondsToSelector:@selector(setAllowsDisplayModeCustomization:)]) {
+            [self.toolbar performSelector:@selector(setAllowsDisplayModeCustomization:) withObject:@(NO)];
         }
         [self.toolbar setAllowsUserCustomization:NO];
         [self.toolbar setSizeMode:NSToolbarSizeModeRegular];
```

Which enabled me to build successfully and run Ladybird. 
This does bypass the compile time check and replaces it with a runtime check. 
It should work in both MacOS 14 and 15 but might not be the desired fix as it is not a compile time check.

--- a/UI/AppKit/Interface/TabController.mm
+++ b/UI/AppKit/Interface/TabController.mm
@@ -94,7 +94,9 @@ static NSString* const TOOLBAR_TAB_OVERVIEW_IDENTIFIER = @"ToolbarTabOverviewIde
         [self.toolbar setDelegate:self];
         [self.toolbar setDisplayMode:NSToolbarDisplayModeIconOnly];
         if (@available(macOS 15, *)) {
-            [self.toolbar setAllowsDisplayModeCustomization:NO];
+            if ([self.toolbar respondsToSelector:@selector(setAllowsDisplayModeCustomization:)]) {
+                [self.toolbar performSelector:@selector(setAllowsDisplayModeCustomization:) withObject:nil];
+            }
         }
         [self.toolbar setAllowsUserCustomization:NO];
         [self.toolbar setSizeMode:NSToolbarSizeModeRegular];

